### PR TITLE
OCPBUGS-19403: updates etcd procedure with OVN-K i/c

### PR DESCRIPTION
**Pull request by JoeAldinger**
Version(s):
4.14+

Issue:
https://issues.redhat.com/browse/OCPBUGS-19403

Link to docs preview:
https://64939--docspreview.netlify.app/openshift-enterprise/latest/backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state#:~:text=If%20you%20are%20using%20the%20OVNKubernetes%20network%20plugin%2C%20you%20must%20restart%20ovnkube%2Dcontrolplane%20pods.

QE review:
- [x] QE has approved this change.

Additional information:
PTAL: @geliu2016 
Looking for another review since I changed so much based on excellent feedback.

### CODE REVIEW

**Comment by palonsoro** [comment]
@JoeAldinger please wherever it says `$<node>` please use `<node>`. It is me (again) using the bash-like syntax, while the docs better benefit from the `<whatever>` syntax.

In new step 13.d, I'd warn that it may take long for the pod to be ready (up to several minutes). Users may get nervous otherwise.

For the rest, it looks perfect to me. Many thanks! Also looking forward for the QE testing, just to be sure that this doesn't break under scenarios I might have not considered.

**Comment by palonsoro** [comment]
It now looks better :slightly_smiling_face:

File: modules/dr-restoring-cluster-state.adoc [hunk 1]
```diff
@@ -281,75 +281,61 @@ etcd-ip-10-0-143-125.ec2.internal                1/1     Running     1
 +
 If the status is `Pending`, or the output lists more than one running etcd pod, wait a few minutes and check again.
 
-. If you are using the `OVNKubernetes` network plugin, delete the node objects that are associated with control plane hosts that are not the recovery control plane host.
+. If you are using the `OVNKubernetes` network plugin, you must restart `ovnkube-controlplane` pods.
+.. Delete all the ovnkube-controlplane pods by running the following command:
```
Comment (bergerhoffer) [inline modules/dr-restoring-cluster-state.adoc:285]: Should `ovnkube-controlplane` be in backticks like the above line has it?  Same for line 291

File: modules/dr-restoring-cluster-state.adoc [hunk 2]
```diff
@@ -281,75 +281,61 @@ etcd-ip-10-0-143-125.ec2.internal                1/1     Running     1
...
+$ oc -n openshift-ovn-kubernetes get pod -l app=ovnkube-control-plane
 ----
-+
-[NOTE]
-====
-It can take at least 5-10 minutes for the OVN-Kubernetes control plane to be redeployed and the previous command to return empty output.
-====
 
-. Restart the Open Virtual Network (OVN) Kubernetes pods on all the hosts.
+. If you are using OVNKubernetes network plugin, restart the Open Virtual Network (OVN) Kubernetes pods on all the nodes one by one. Steps to restart OVN-K pods in each node are as follows:
```
Comment (bergerhoffer) [inline modules/dr-restoring-cluster-state.adoc:298]: * To be consistent with how it's written above, do "If you are using the `OVNKubernetes` network plugin"
* Is "OVN-K" an approved shortened form, or should it be written out?
* Consider rewording the last sentence to be more active

```suggestion
. If you are using the `OVNKubernetes` network plugin, restart the Open Virtual Network (OVN) Kubernetes pods on all the nodes one by one. Use the following steps to restart OVN-K pods on each node:
```

File: modules/dr-restoring-cluster-state.adoc [hunk 3]
```diff
@@ -281,75 +281,61 @@ etcd-ip-10-0-143-125.ec2.internal                1/1     Running     1
...
 +
 [NOTE]
 ====
-Validating and mutating admission webhooks can reject pods. If you add any additional webhooks with the `failurePolicy` set to `Fail`, then they can reject pods and the restoration process can fail. You can avoid this by saving and deleting webhooks while restoring the cluster state. After the cluster state is restored successfully, you can enable the webhooks again.
-
-Alternatively, you can temporarily set the `failurePolicy` to `Ignore` while restoring the cluster state. After the cluster state is restored successfully, you can set the `failurePolicy` to `Fail`.
+Restart OVN-K pods in the following order: the recovery control plane host, the other control plane hosts (if available), and lastly the other nodes.
```
Comment (bergerhoffer) [inline modules/dr-restoring-cluster-state.adoc:302]: * Same question about OVN-K, should it be "OVN-Kubernetes"?
* Since this is a specific order, I'd consider making an actual ordered list

File: modules/dr-restoring-cluster-state.adoc [hunk 4]
```diff
@@ -281,75 +281,61 @@ etcd-ip-10-0-143-125.ec2.internal                1/1     Running     1
...
+.. Restart the OpenVSwitch services. Access the node by using Secure Shell (SSH) and run the following command:
 +
 [source,terminal]
 ----
-$ oc delete pods -l app=ovnkube-master -n openshift-ovn-kubernetes
+$ sudo systemctl restart ovs-vswitchd ovsdb-server
 ----
 
-.. Ensure that any OVN-Kubernetes control plane pods are deployed again and are in a `Running` state by running the following command:
+.. Delete the `ovnkube-node` pod running on the node by running the following command:
```
Comment (bergerhoffer) [inline modules/dr-restoring-cluster-state.adoc:319]: This uses "running" twice in the same sentence, with two different contexts. I would suggest rewording to avoid that. You can probably just remove the first instance of "running" and it would keep the same meaning

File: modules/dr-restoring-cluster-state.adoc [hunk 5]
```diff
@@ -281,75 +281,61 @@ etcd-ip-10-0-143-125.ec2.internal                1/1     Running     1
...
-.Example output
-[source,terminal]
-----
-NAME                   READY   STATUS    RESTARTS   AGE
-ovnkube-master-nb24h   4/4     Running   0          48s
-----
+<1> Replace `<node>` with the name of the node you are restarting.
 
-.. Delete all `ovnkube-node` pods by running the following command:
+.. Verifty that the `ovnkube-node` pod is running again. Run the following command:
```
Comment (bergerhoffer) [inline modules/dr-restoring-cluster-state.adoc:328]: * Typo Verifty -> Verify
* I would combine the sentences "... by running the following command". However that does leave you in the same place as above with two "running"'s in there. Maybe you could reword though to avoid (that may have been why you broke it up in to two sentences, but there could be another way)

File: modules/dr-restoring-cluster-state.adoc [hunk 6]
```diff
@@ -281,75 +281,61 @@ etcd-ip-10-0-143-125.ec2.internal                1/1     Running     1
...
-
-.. Ensure that all the `ovnkube-node` pods are deployed again and are in a `Running` state by running the following command:
 +
-[source,terminal]
-----
-$ oc get  pods -n openshift-ovn-kubernetes | grep ovnkube-node
-----
+[NOTE]
+====
+It may take several minutes for the pods to restart.
```
Comment (bergerhoffer) [inline modules/dr-restoring-cluster-state.adoc:337]:
```suggestion
It might take several minutes for the pods to restart.
```

File: modules/dr-restoring-cluster-state.adoc [hunk 7]
```diff
@@ -281,75 +281,61 @@ etcd-ip-10-0-143-125.ec2.internal                1/1     Running     1
...
+$ oc -n openshift-ovn-kubernetes delete pod -l app=ovnkube-control-plane
 ----
-
-. Verify that the Cluster Network Operator (CNO) redeploys the OVN-Kubernetes control plane and that it no longer references the non-recovery controller IP addresses. To verify this result, regularly check the output of the following command. Wait until it returns an empty result before you proceed to restart the Open Virtual Network (OVN) Kubernetes pods on all of the hosts in the next step.
+.. Deploy all of the ovnkube-controlplane pods again by running the following command:
 +
 [source,terminal]
 ----
-$ oc -n openshift-ovn-kubernetes get ds/ovnkube-master -o yaml | grep -E '<non-recovery_controller_ip_1>|<non-recovery_controller_ip_2>'
+$ oc -n openshift-ovn-kubernetes get pod -l app=ovnkube-control-plane
```
Comment (bergerhoffer) [inline modules/dr-restoring-cluster-state.adoc:295]: Definitely confirm with an SME, but I don't think this command will deploy the pods again. The `get` command just retrieves information.

Though actually now that I think about it, they are probably redeployed automatically because you deleted the pods in the previous command. So you might just need to reword the above step to not say "Deploy", but instead say something like verify that all of the pods are deployed again...

**Review by bergerhoffer — COMMENTED** [review]
A few suggestions!

File: modules/dr-restoring-cluster-state.adoc (hunk 3 above)
Comment (palonsoro) [reply modules/dr-restoring-cluster-state.adoc:302]: @JoeAldinger now that I re-read this: The paragraph that you are deleting with this change should not be deleted and you are deleting it. Please keep it.

**Comment by palonsoro** [comment]
@JoeAldinger re-posting here because, for some reason, when I write on reviews, github sometimes doesn't forward my comments:

Re-reading, I saw that you deleted the following paragraphs
```
Validating and mutating admission webhooks can reject pods. If you add any additional webhooks with the `failurePolicy` set to `Fail`, then they can reject pods and the restoration process can fail. You can avoid this by saving and deleting webhooks while restoring the cluster state. After the cluster state is restored successfully, you can enable the webhooks again.
```
and
```
Alternatively, you can temporarily set the `failurePolicy` to `Ignore` while restoring the cluster state. After the cluster state is restored successfully, you can set the `failurePolicy` to `Fail`.
```

**Please don't delete them**. They are the solution for a previous bug, so it would be a regression.

Sorry that I didn't notice before.

File: modules/dr-restoring-cluster-state.adoc (hunk 7 above)
Comment (palonsoro) [reply modules/dr-restoring-cluster-state.adoc:295]: It should be `delete`, not `get`. Good catch.

**Comment by palonsoro** [comment]
As said in the review, in case the comment doesn't make it: command at newer line 295 should be "delete", not "get".

File: modules/dr-restoring-cluster-state.adoc (hunk 7 above)
Comment (bergerhoffer) [reply modules/dr-restoring-cluster-state.adoc:295]: @palonsoro The previous command (line 289) does already delete the pods.

This command looks like it might just be trying to verify that the pods were redeployed. So it might just be the wording above the command that needs to be changed (instead of saying "Deploy... " maybe it should say something about verifying that the pods were redeployed?)

Comment (palonsoro) [reply modules/dr-restoring-cluster-state.adoc:295]: Ok. I missed the context around. Then yes, you don't deploy but you wait for the pods to be deployed (by OpenShift)

File: modules/dr-restoring-cluster-state.adoc [hunk 8]
```diff
@@ -281,26 +281,29 @@ etcd-ip-10-0-143-125.ec2.internal                1/1     Running     1
...
 ----
+
+. If you are using the `OVNKubernetes` network plugin, restart the Open Virtual Network (OVN) Kubernetes pods on all the nodes one by one. Use the following steps to restart OVN-Kubernetes pods on each node:
 +
-[NOTE]
+[IMPORTANT]
 ====
-It can take at least 5-10 minutes for the OVN-Kubernetes control plane to be redeployed and the previous command to return empty output.
```
Comment (JoeAldinger) [inline modules/dr-restoring-cluster-state.adoc:300]: @palonsoro does this need to be added back in as well?

Comment (palonsoro) [reply modules/dr-restoring-cluster-state.adoc:300]: I don't think so. The ovn-kubernetes pods are now much more lightweight. This sentence is present in 4.13 and older because CNO has to re-deploy the control plane single replica and it intentionally waits several minutes before doing so. That's no longer the case.

File: modules/dr-restoring-cluster-state.adoc [hunk 9]
```diff
@@ -281,26 +281,29 @@ etcd-ip-10-0-143-125.ec2.internal                1/1     Running     1
...
-. Verify that the Cluster Network Operator (CNO) redeploys the OVN-Kubernetes control plane and that it no longer references the non-recovery controller IP addresses. To verify this result, regularly check the output of the following command. Wait until it returns an empty result before you proceed to restart the Open Virtual Network (OVN) Kubernetes pods on all of the hosts in the next step.
+.. Verify all of the `ovnkube-controlplane` pods were redeployed by running the following command:
 +
 [source,terminal]
 ----
-$ oc -n openshift-ovn-kubernetes get ds/ovnkube-master -o yaml | grep -E '<non-recovery_controller_ip_1>|<non-recovery_controller_ip_2>'
+$ oc -n openshift-ovn-kubernetes get pod -l app=ovnkube-control-plane
 ----
+
+. If you are using the `OVNKubernetes` network plugin, restart the Open Virtual Network (OVN) Kubernetes pods on all the nodes one by one. Use the following steps to restart OVN-Kubernetes pods on each node:
```
Comment (kowen-rh) [inline modules/dr-restoring-cluster-state.adoc:298]: Is there a reason that `OVNKubernetes` is in backticks, but "OVN-Kubernetes" is not? Are these different things?

File: modules/dr-restoring-cluster-state.adoc [hunk 10]
```diff
@@ -313,43 +316,36 @@ Alternatively, you can temporarily set the `failurePolicy` to `Ignore` while res
...
+$ sudo systemctl restart ovs-vswitchd ovsdb-server
 ----
 
-.. Ensure that any OVN-Kubernetes control plane pods are deployed again and are in a `Running` state by running the following command:
+.. Delete the `ovnkube-node` pod on the node by running the following command:
 +
 [source,terminal]
 ----
-$ oc get pods -l app=ovnkube-master -n openshift-ovn-kubernetes
+$ oc -n openshift-ovn-kubernetes delete pod -l app=ovnkube-node --field-selector=spec.nodeName==<node> <1>
```
Comment (kowen-rh) [inline modules/dr-restoring-cluster-state.adoc:333]: Is a callout necessary here? I feel like we can reword the instructions to avoid using a single callout in a single command with something like this:

```adoc
.. Delete the `ovnkube-node` pod on the node by running the following command, replacing `<node>` with the name of the node that you are restarting:
```

**Review by kowen-rh — COMMENTED** [review]
A few minor nitpicks, but overall this looks good to me!

/remove-label peer-review-in-progress
/label peer-review-done

File: modules/dr-restoring-cluster-state.adoc (hunk 9 above)
Comment (JoeAldinger) [reply modules/dr-restoring-cluster-state.adoc:298]: No just a mistake.

File: modules/dr-restoring-cluster-state.adoc (hunk 10 above)
Comment (JoeAldinger) [reply modules/dr-restoring-cluster-state.adoc:333]: I agree. More importantly I think this will create some issues with copying the code.
